actors/miner/src/lib.rs

```diff
@@ -1950,6 +1951,116 @@ impl Actor {
+        params: ProveCommitSectorsNIParams,
+    ) -> Result<(), ActorError> {
+        info!("Test: prove_commit_sectors_ni");
+
+        let state: State = rt.state()?;
+        let store = rt.store();
+        let policy = rt.policy();
+        let miner_id = rt.message().receiver().id().unwrap();
+        let info = get_miner_info(rt.store(), &state)?;
+        let activation_epoch = rt.curr_epoch();
+
+        rt.validate_immediate_caller_is(
+            info.control_addresses.iter().chain(&[info.worker, info.owner]),
+        )?;
+
+        let miner_id = rt.message().receiver().id().unwrap();
+
+        let mut sectors_to_add = Vec::new();
+        let mut total_pledge = TokenAmount::zero();
+
+        for (i, sector) in params.sectors.iter().enumerate() {
+            let seal_verify_info = NISealVerifyInfo {
+                registered_proof: params.seal_proof_type,
+                sector_id: fvm_shared::sector::SectorID {
+                    miner: miner_id,
+                    number: sector.sector_number,
+                },
+                randomness: Randomness(Vec::new()),
+                proof: params.sector_proofs[i].clone().into(),
+                sealed_cid: sector.sealed_cid,
+                unsealed_cid: CompactCommD::empty().get_cid(params.seal_proof_type).unwrap(),
+            };
+            rt.batch_verify_ni_seals(&vec![seal_verify_info])
+                .context_code(ExitCode::USR_ILLEGAL_ARGUMENT, "failed to batch verify")?;
+
+            let duration = sector.expiration - activation_epoch;
+            let power =
+                qa_power_for_weight(info.sector_size, duration, &BigInt::zero(), &BigInt::zero());
+
+            let rew = request_current_epoch_block_reward(rt)?;
+            let pwr = request_current_total_power(rt)?;
+            let circulating_supply = rt.total_fil_circ_supply();
+            let pledge_inputs = NetworkPledgeInputs {
+                network_qap: pwr.quality_adj_power_smoothed,
+                network_baseline: rew.this_epoch_baseline_power,
+                circulating_supply,
+                epoch_reward: rew.this_epoch_reward_smoothed,
+            };
+
+            let storage_pledge = expected_reward_for_power(
+                &pledge_inputs.epoch_reward,
+                &pledge_inputs.network_qap,
+                &power,
+                INITIAL_PLEDGE_PROJECTION_PERIOD,
+            );
+
+            let initial_pledge = initial_pledge_for_power(
+                &power,
+                &pledge_inputs.network_baseline,
+                &pledge_inputs.epoch_reward,
+                &pledge_inputs.network_qap,
+                &pledge_inputs.circulating_supply,
+            );
+
+            let day_reward = expected_reward_for_power(
+                &pledge_inputs.epoch_reward,
+                &pledge_inputs.network_qap,
+                &power,
+                fil_actors_runtime::EPOCHS_IN_DAY,
+            );
+            total_pledge += &initial_pledge;
```

Comment:
Knowing how this is calculated, it will be exactly the same for every sector, regardless of their duration. So we can pre-compute all this stuff once before the iteration.